### PR TITLE
Add libblkio release test

### PR DESCRIPTION
**Pull request by qingwangrh**
It will verify the source code of
libblkio is working and test the main
driver virtio-blk-vhost-vdpa

ID:2222155

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=block_libblkio_release --guestname=RHEL.9.3.0 --clone=no --firmware=default_bios --nrepeat=5

 (1/5) repeat1.Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: STARTED
 (1/5) repeat1.Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (69.05 s)
 (2/5) repeat2.Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: STARTED
 (2/5) repeat2.Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (68.50 s)
 (3/5) repeat3.Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: STARTED
 (3/5) repeat3.Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (68.17 s)
 (4/5) repeat4.Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: STARTED
 (4/5) repeat4.Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (69.07 s)
 (5/5) repeat5.Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: STARTED
 (5/5) repeat5.Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (68.55 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by qingwangrh** [comment]
@XueqiangWei @zhenyzha @fbq815 @maxujun Please help to review, thanks.

**Comment by zhenyzha** [comment]
git am 3780.patch
Applying: Add libblkio release test
.git/rebase-apply/patch:44: new blank line at EOF.
+
warning: 1 line adds whitespace errors.

**Comment by qingwangrh** [comment]
> git am 3780.patch Applying: Add libblkio release test .git/rebase-apply/patch:44: new blank line at EOF. + warning: 1 line adds whitespace errors.

updated

File: qemu/tests/block_libblkio_release.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    vm.verify_alive()
+
+    session = vm.wait_for_login(timeout=360)
+    logger.info("Copy script to guest")
+    deps_dir = virttest_data_dir.get_deps_dir()
+    host_file = os.path.join(deps_dir, host_script)
+    vm.copy_files_to(host_file, guest_dir)
+
+    logger.info("Execute script in guest")
+    guest_cmd_output = session.cmd(guest_cmd, timeout=360)
```
Comment (XueqiangWei) [inline qemu/tests/block_libblkio_release.py:32]: Suggest to use "session.cmd_status_output". Thanks.

**Comment by XueqiangWei** [comment]
(1/5) repeat1.Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: STARTED
 (1/5) repeat1.Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (210.49 s)
 (2/5) repeat2.Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: STARTED
 (2/5) repeat2.Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (79.14 s)
 (3/5) repeat3.Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: STARTED
 (3/5) repeat3.Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (77.80 s)
 (4/5) repeat4.Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: STARTED
 (4/5) repeat4.Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (78.80 s)
 (5/5) repeat5.Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: STARTED
 (5/5) repeat5.Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (80.01 s)

**Comment by maxujun** [comment]
(1/1) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.block_libblkio_release: STARTED
 (1/1) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.block_libblkio_release: CANCEL: Got host qemu version:6.2.0-36, which is not in [8.0.0,) (2.93 s)

LGTM, Acked.

**Comment by zhenyzha** [comment]
aarch64 doesn't have this module in kvm, please skip aarch64 for this mr too.

File: qemu/tests/block_libblkio_release.py (hunk 1 above)
Comment (qingwangrh) [reply qemu/tests/block_libblkio_release.py]: updated

**Comment by XueqiangWei** [comment]
(1/1) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.block_libblkio_release.q35: PASS (85.45 s)

**Review by XueqiangWei — APPROVED** [review]

**Comment by SiboWang1997** [comment]
Hi @XueqiangWei,  aarch64 and s390x don't have this module in kvm, but it seems that they are not filtered out, could you review it again?

**Comment by SiboWang1997** [comment]
@zhencliu @YongxueHong

**Comment by XueqiangWei** [comment]
> Hi @XueqiangWei, aarch64 and s390x don't have this module in kvm, but it seems that they are not filtered out, could you review it again?

Thank you for pointing out it, since this pullreq has been merged. Let's try it in a new pullreq. cc @qingwangrh

**Comment by wq3721** [comment]
> Hi @XueqiangWei, aarch64 and s390x don't have this module in kvm, but it seems that they are not filtered out, could you review it again?
> 
> Cc: @zhencliu @YongxueHong

suggest filter out in kar loop.

**Comment by XueqiangWei** [comment]
> > Hi @XueqiangWei, aarch64 and s390x don't have this module in kvm, but it seems that they are not filtered out, could you review it again?
> > Cc: @zhencliu @YongxueHong
> 
> suggest filter out in kar loop.

Yes, @qingwangrh is more inclined to use this way. He has discussed it with @zhenyzha and will discuss it with @SiboWang1997. Thank you.

**Pull request merged**